File: erpnext/manufacturing/report/selling_price_report/selling_price_report.js
// Copyright (c) 2016, Frappe Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt

frappe.query_reports["Selling Price Report"] = {
	"filters": [
		{
			"fieldname":"date",
			"label": ("Date"),
			"fieldtype": "Date",
			"reqd": 1
		},
		// {
		// 	"fieldname":"to_date",
		// 	"label": ("Date"),
		// 	"fieldtype": "Date",
		// 	"reqd": 1
		// },
		{
			"fieldname": "branch",
            "label": ("Branch"),
        	"fieldtype": "Link",
            "options": "Branch",
            // "get_query": function() {
			// 		return {"doctype": "Branch", "filters": {"is_disabled": 0}}
			// },
			 "reqd":1
		},
		{
			"fieldname":"item",
			"label": ("Product"),
			"fieldtype": "Link",
			"options": "Item",
			// "get_query": function() {
			// 	var branch = frappe.query_report.filters_by_name.branch.get_value();
			// 	if(branch)
			// 		return {"doctype": "Employee", "filters": {"branch": branch }} 
			// 	}	
		}

	]
}

File: erpnext/manufacturing/report/selling_price_report/selling_price_report.py
# Copyright (c) 2013, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

#########Created by Cheten Tshering on 17/09/2020 ##################################
from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, data = get_columns(filters),get_data(filters)
	return columns, data

def get_columns(filters):
	columns = [
		_("Product Code") + ":Link/Item:150",
		_("Product Name") + ":Data:150",
		_("BOM") + ":Link/BOM:150",
		_("BOM Cost") + ":Currency:150",
		_("Prime Cost") + ":Currency:150",
		_("Manufacturing Cost") + ":Currency:150",
		_("Cost of Production") + ":Currency:150",
		_("Selling Price") + ":Currency:150",
		_("Cost Sheet") + "::150"

	]
	return columns

def get_data(filters):
	cond = get_conditions(filters)
	data = frappe.db.sql("""
		select 
			cs.item, 
			cs.item_name, 
			cs.bom, 
			cs.bom_cost, 
			cs.prime_cost, 
			cs.manufacturing_cost, 
			cs.production_cost, 
			cs.selling_price, 
			cs.name
		from `tabCost Sheet` cs
		where cs.docstatus = 1 
		and '{date}' >= cs.from_date
		{condition}
	""".format(date=filters.date, condition=cond))
	return data

def get_conditions(filters):
	cond = ""
	if filters.branch:
		cond += """ and branch = "{}" """.format(filters.branch)
	if filters.item:
		cond += """and item = "{}" """.format(filters.item)
	return cond